content/articles/2018-05-dplyr-0-7-5.Rmd

```diff
@@ -0,0 +1,161 @@
+
+The next planned release of *dplyr* will be a feature release: Romain, author of the data frame backend for *dplyr* and coauthor of the *Rcpp* package, has joined the team and hit the ground running.  He has implemented many of the features described in this blog post, many more are already available in the development version.  Welcome (back), Romain, looking forward to a great time!
+
+
+## tidyselect
+
+*dplyr* always supported selecting by name, excluding columns, selecting by range, by match, or by position:
+
+```{r select}
+tbl <- data.frame(a1 = 1, a2 = 2, a3 = 3, b = "x")
+
+tbl %>%
+  select(a1, a2)
+tbl %>%
+  select(-b)
+tbl %>%
+  select(a1:a3)
+tbl %>%
+  select(starts_with("a"))
+tbl %>%
+  select(2:4)
+vars <- syms(c("a2", "b"))
+tbl %>%
+  select(!!!vars)
+```
+
+Last year, the core code that provides this functionality out of *dplyr* into the fairly new *tidyselect* package.  Selecting columns in a data frame (or items in a character vector, for that matter) is a common task in many other situations.  The *tidyselect* package offers a consistent and convenient interface with full support for quasiquotation, and is used by more than 20 packages, and now also by *dplyr*.  Internally, the `select()` calls above are translated into the following *tidyselect* operations:
+
+```{r tidyselect}
+tbl_names <- names(tbl)
+
+tbl_names %>%
+  tidyselect::vars_select(a1, a2)
+tbl_names %>%
+  tidyselect::vars_select(-b)
+tbl_names %>%
+  tidyselect::vars_select(a1:a3)
+tbl_names %>%
+  tidyselect::vars_select(starts_with("a"))
+tbl_names %>%
+  tidyselect::vars_select(2:4)
+vars <- syms(c("a2", "b"))
+tbl_names %>%
+  tidyselect::vars_select(!!!vars)
+```
+
+The net effect of this change is improved consistency across the tidyverse and the other packages that use *tidyselect*.  The user interface is affected in two ways:
+
+- The `select_vars()`, `select_var()` and `rename_vars()` functions are soft-deprecated and will start issuing warnings in a future version. Instead, use `tidyselect::vars_select()`, `tidyselect::vars_pull()` and `tidyselect::rename_vars()`, respectively.
+
+- `select()` and `rename()` fully support character vectors. You can now unquote variables like this:
+
+    ```{r select-char}
+    vars <- c("a2", "b")
+    select(tbl, !!vars)
+    select(tbl, -(!!vars))
+    ```
+
+## Scoped verbs for grouped data
+
+Scoped verbs are useful when you want to apply the same operation on multiple columns.  These functions end with `_all` (affect all columns), `_at` (affect selected columns), or `_if` (affect columns that satisfy a predicate), and replaced the older `_each` family of functions in *dplyr* 0.7.0.  In the most recent version of *dplyr*, these functions have been extended to work on grouped data frames.  Because the group columns need to stay unchanged, these operations work slightly differently on grouped data frames:
+
+- In `select_...()`, the group columns are always kept.
```

Comment:
For typed map functions in purrr, I believe we use `map_*()`, would it make sense to do the same for `select_*()` etc.?